crypto/Makefile

```diff
@@ -70,12 +70,13 @@ ZKP_CFLAGS = \
 	-DENABLE_MODULE_SCHNORRSIG \
 	-DENABLE_MODULE_EXTRAKEYS
 ZKP_PATH = ../vendor/secp256k1-zkp
+# this is specific for 64-bit builds
 CFLAGS += -DSECP256K1_CONTEXT_SIZE=208
 
 # TODO remove this workaround once possible
-ifeq ($(CLANG_AT_LEAST_14),true)
-$(warning "warning: suppressing clang-14 compiler warning for secp256k1-zkp code")
-ZKP_CFLAGS += -Wno-bitwise-instead-of-logical
+ifeq ($(CLANG_AT_LEAST_15),true)
+$(warning "warning: suppressing clang-15 `-Wstrict-prototypes` compiler warning for `zkp_context.c` as workaround")
+zkp_context.o: CFLAGS += -Wno-strict-prototypes
```

Comment:
Why do we need `no-strict-prototypes`? I don't have clang-15. Can you paste the compiler warning here?
If it's just a matter of changing `int zkp_context_init()` -> `int zkp_context_init(void)` etc., then I think we should fix the function definitions.

Reply:
For the crypto fuzzer, I often work around other code issues to move ahead and introduce as few controversial or incomplete changes to the main code as possible. 

When building the crypto fuzzer with clang, only the instances in `zkp_context.c` are seen by the compiler, but there are other instances in other files that should ideally get fixed as well. That's why I opened https://github.com/trezor/trezor-firmware/issues/2360 .

@andrewkozlik I see three options:
1) we stick with the Makefile workaround and fix the issues as part of a different PR
1) I fix `zkp_context.c` as part of this PR and drop the Makefile workaround
1) I fix all instances of this issue as part of this PR and drop the Makefile workaround

Which option do you prefer?

Reply:
Compiler warning on clang-15 and early clang-16:
```
zkp_context.c:54:21: error: a function declaration without a prototype is deprecated in all versions of C [-Werror,-Wstrict-prototypes]
int zkp_context_init() {
                    ^
                     void
zkp_context.c:81:25: error: a function declaration without a prototype is deprecated in all versions of C [-Werror,-Wstrict-prototypes]
void zkp_context_destroy() {
                        ^
                         void
zkp_context.c:90:51: error: a function declaration without a prototype is deprecated in all versions of C [-Werror,-Wstrict-prototypes]
const secp256k1_context *zkp_context_get_read_only() {
                                                  ^
                                                   void
zkp_context.c:97:48: error: a function declaration without a prototype is deprecated in all versions of C [-Werror,-Wstrict-prototypes]
secp256k1_context *zkp_context_acquire_writable() {
                                               ^
                                                void
zkp_context.c:108:34: error: a function declaration without a prototype is deprecated in all versions of C [-Werror,-Wstrict-prototypes]
void zkp_context_release_writable() {
                                 ^
                                  void
```

Reply:
I suggest Option 2.

When I try to build with clang 14.0.0 I get the error below. It's interesting that this doesn't trouble newer versions of the compiler.
```
fuzzer/fuzzer.c: In function ‘fuzz_ecdsa_recover_pub_from_sig_functions’:
fuzzer/fuzzer.c:1020:11: error: variable ‘zero_pubkey’ set but not used [-Werror=unused-but-set-variable]
 1020 |   uint8_t zero_pubkey[65] = {0};
      |           ^~~~~~~~~~~
```

Reply:
@andrewkozlik : that's curious, I cannot get my clang versions to detect the dead store even with `-Wunused-but-set-variable` and `-Wunused-variable` explicitly enabled. `zero_pubkey` is definitely an unused leftover that I should remove, so it's unexpected that it is not found.

Could you list the exact commands and flags that you used during your build?

Reply:
https://github.com/trezor/trezor-firmware/pull/2449/commits/2c5fcf1ab4c011d9f186de5350f4e759f147ab98 removes the unused variable and some other dead stores.

Reply:
I was wrong, it was actually GCC showing the error. I thought `make fuzzer` automatically uses clang as the default compiler.

Reply:
That explains it. Although it's currently not important that GCC can build the fuzzer code since it lacks the fundamental `-fsanitize=fuzzer` libFuzzer support, it's still useful to look at what it complains about. As we've seen, it sometimes catches issues that clang doesn't find.

Therefore I've added https://github.com/trezor/trezor-firmware/pull/2449/commits/dda39e7fc684004334f68226f369a8df676d3cc9 to resolve the remaining code-specific warnings.

I'm aware that `FUZZER=1 make fuzzer` is not very intuitive (although it is [documented](https://github.com/trezor/trezor-firmware/blob/master/crypto/fuzzer/README.md#build)) and am open to suggestions on how to solve the conditional changes to the build more elegantly. In my understanding, target-specific Makefile variables can make the Makefile evaluation more complex if they have a global impact, so I've relied on external configuration of `FUZZER=1` for now that is straightforward to reason about.